google/cloud/storage/internal/unified_rest_credentials.cc

```diff
@@ -69,14 +69,19 @@ class WrapRestCredentials : public oauth2::Credentials {
   std::shared_ptr<oauth2_internal::Credentials> impl_;
 };
 
-struct RestVisitor : public CredentialsVisitor {
+class RestVisitor : public CredentialsVisitor {
```

Comment:
Should we make `RestVisitor` local to `MapCredentials()` (just like `Visitor` was in the non-storage version) so that it is clearer that any one instance can't be visited twice (because the factory is consumed)?